src/main/scala/com/mozilla/telemetry/monitoring/DogStatsDMetric.scala

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package com.mozilla.telemetry.monitoring
+
+sealed trait DogStatsDMetric {
+  val metricName: String
+  protected val metricStringValue: String
+  protected val metricType: String
+  val kvTags: Option[Map[String, String]]
+  val bareTags: Option[Seq[String]]
+
+  def format(sampleRate: Option[Double] = None): String = {
+    Array(Some(metric), Some(metricType), sampleRateString(sampleRate), tags)
+      .flatten
+      .mkString("|")
+  }
+
+  private def normalize(input: String): String = {
+    input.replaceAll("[:|@]", "_")
+  }
+
+  private def sampleRateString(sampleRate: Option[Double]): Option[String] = sampleRate.map("@" + _.toString)
```

Comment:
nit: I'd add a comment pointing to Datadog documentation to explain magic `@` here and others in line 20 (https://github.com/mozilla/telemetry-streaming/pull/167/files#diff-ebf9e4f9305826d7c4e93f152216a9e0R20)

Reply:
Makes sense -- I added a link to the docs at the top of the trait and added a comment on the `normalize` method about the reserved characters